samza-core/src/main/java/org/apache/samza/processor/StreamProcessor.java

```diff
@@ -255,20 +259,25 @@ public void onContainerStart() {
                 processorListener.onStart();
               }
             } else {
-              LOGGER.warn("Received duplicate container start notification for container: {} in stream processor: {}.", container, processorId);
+              LOGGER.warn("Received duplicate container start notification for container: {} in stream processor: {}.", containerRef.get(), processorId);
             }
           }
 
           @Override
           public void onContainerStop(boolean pauseByJm) {
             if (pauseByJm) {
-              LOGGER.info("Container: {} of the stream processor: {} was stopped by the JobCoordinator.", container, processorId);
+              SamzaContainer container = containerRef.get();
+              if (container != null) {
+                LOGGER.info("Container: {} of the stream processor: {} was stopped by the JobCoordinator.", container, processorId);
+              }
               if (jcContainerShutdownLatch != null) {
                 jcContainerShutdownLatch.countDown();
               }
             } else {  // sp.stop was called or container stopped by itself
-              LOGGER.info("Container: {} stopped. Stopping the stream processor: {}.", container, processorId);
-              container = null; // this guarantees that stop() doesn't try to stop container again
+              SamzaContainer container = containerRef.getAndSet(null); // this guarantees that stop() doesn't try to stop container again
+              if (container != null) {
+                LOGGER.info("Container: {} stopped. Stopping the stream processor: {}.", container, processorId);
```

Comment:
Would it be useful to log the `processorId` even though `container` is null? Same goes for the if block above.